### PR TITLE
Consume client-side javascript via NPM instead of local files

### PR DESCRIPTION
**Pull request by igloosi**
At the moment we have libraries that are needed for certain functionality saved locally in the source/ folder
For the sake of consistency and flexibility we've removed those and consumed them via npm.
This PR:
* replaces local `Clipboard.js` with a version from NPM:
    * adds it's path in node_modules to `config.paths`
    * adds a step in `metalsmith.js` that copies that file to `deploy/public/vendor`
    * deletes local version

* replaces local `jQuery.js` with a version from NPM (1.12.4):
    * adds it's path in node_modules to `config.paths`
    * adds a step in `metalsmith.js` that copies that file to `deploy/public/vendor`
    * deletes local version

* replaces local `html5shiv-printshiv.js` with a version from NPM:
    * adds it's path in node_modules to `config.paths`
    * adds a step in `metalsmith.js` that copies that file to `deploy/public/vendor`
    * deletes local version

* replaces local `modernizr.js` custom build with a version from NPM 
    * create a a config file in `config/modernizr.json`for features of modernizr needed
    * creates a `modernizr-build.js` plugin in `lib/`that builds the custom modernizr build to deploy folder
    * adds [`fs-extra`](https://github.com/jprichardson/node-fs-extra) for simplified async promise support for fs events 
    * adds a `.use(modernizrBuild({` step in `metalsmith.js` that triggers the build
    * deletes local version

Trello ticket: https://trello.com/c/aJXT3AES/552-3-update-ds-to-consume-client-side-javascript-using-npm

### CODE REVIEW

**Comment by igloosi** [comment]
hmm, related to npm test user password change?
```
10:54:33 AM: Installing NPM modules using NPM version 5.6.0
10:54:34 AM: > govuk_design_system@ preinstall /opt/build/repo
10:54:34 AM: > node bin/check-nvmrc.js
10:54:37 AM: npm
10:54:37 AM:  WARN tar ENOENT: no such file or directory, open '/opt/build/repo/node_modules/.staging/nunjucks-034b6144/contribute.json'
10:54:37 AM: npm WARN tar ENOENT: no such file or directory, open '/opt/build/repo/node_modules/.staging/nunjucks-034b6144/.jshintrc'
10:54:38 AM: npm ERR!
10:54:38 AM:  code E404
10:54:38 AM: npm ERR! 404 Not Found: @govuk-frontend/icons@https://registry.npmjs.org/@govuk-frontend/icons/-/icons-0.0.22-alpha.tgz
10:54:38 AM: npm ERR! A complete log of this run can be found in:
10:54:38 AM: npm ERR!     /opt/buildhome/.npm/_logs/2018-02-05T10_54_38_703Z-debug.log
10:54:38 AM: Error during NPM install
```

File: lib/file-helper.js [hunk 1]
```diff
@@ -73,3 +73,23 @@ exports.getDirectories = itemPath => {
   }
   return directories
 }
+
+// This helper function takes a destination path and
+// returns creates all directories for that path in they're not present
+exports.createDirectoriesForPath = (targetDir, {isRelativeToScript = false} = {}) => {
```
Comment (nickcolley) [inline lib/file-helper.js]: Do we need to make this sync? Looks like the place we're using this function would allow for an async implementation of this.

Comment (igloosi) [reply lib/file-helper.js]: wouldn't the file likely be tried to be create before the parent directories are and throeing an error in that case?

Comment (alex-ju) [reply lib/file-helper.js]: the way the function was written it must be sync; if we convert it to async (`mkdir`) we should use callbacks

**Comment by igloosi** [comment]
updated

**Comment by igloosi** [comment]
@alex-ju how does async look to you now?

File: lib/modernizr-build.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+ *
+ * @return {Function}
+ */
+
+const plugin = (opts) => {
+  opts = opts || {}
+  let buildConfig = require(opts.config)
+  return function (files, metalsmith, done) {
+    modernizr.build(buildConfig, (content) => {
+      fs.ensureDir(path.join(paths.public, opts.destination), err => {
```
Comment (alex-ju) [inline lib/modernizr-build.js]: ensures that the directory exists, if the directory structure does not exist, it is created. same as `mkdir -p`. maybe it's worth adding a comment above.

Comment (igloosi) [reply lib/modernizr-build.js]: added

**Pull request merged**